libzkbob-rs/src/merkle.rs

```diff
@@ -346,6 +500,35 @@ impl<D: KeyValueDB, P: PoolParams> MerkleTree<D, P> {
         }
     }
 
+    // Partial tree support
+    // To build Merkle tree from the specified index we should generate left siblings
+    // from the commitment height (in case of index is a multiple of constants::OUT + 1) to the root height
+    // This method should be used on the relayer side to provide the client needed siblings
+    pub fn get_left_siblings(&self, index: u64) -> Option<Vec<Node<P::Fr>>> {
```

Comment:
> why bother with Option? we could just return empty Vec, that would let us iterate it over without pattern matching here

I cover an error case in the Option because the empty Vec is formally a valid case (e.g. this method returns an empty Vec for index 0). But if we requested left siblings for index not within the tree - the method should returns None which indicating the siblings cannot be retrieved from the current state